search_box: Use 'zulip-icon-close' instead of 'fa-remove' in all search boxes.

reharsh opened the pull request:
This is a follow-up PR for #32594, It replaces all '**x**' -`fa-remove` icons with `zulip-icon-close` icons
across all search boxes in the app. 

Additionally, it resolves some hovering and displaying issues, ensuring consistency.

Fixes: #32598 

### Screenshots

|                                    |                                     |
|-------------------------------------------|-------------------------------------------|
| Recent Conversations  | ![recent](https://github.com/user-attachments/assets/439911fa-1e88-49e2-95fd-e7c83b79e919)    |
| User Groups                         | ![Changes visible in alignment and hover](https://github.com/user-attachments/assets/639495b5-e275-49ee-9b69-3ac1a51e1c87)    |
| Direct Messages Left Sidebar  | ![dm](https://github.com/user-attachments/assets/846f4ee2-06bb-4a46-abc7-3f27e4cd2fc6) |
| Channels Settings | ![dm](https://github.com/user-attachments/assets/6861fe59-69e8-46f8-b1cd-0caf6222433a) |
| Channels Left Sidebar | ![dm](https://github.com/user-attachments/assets/323c0899-93b4-4170-97fa-91f940838a5c) |
| User Profile Modal | ![dm](https://github.com/user-attachments/assets/9223677a-8f49-4297-a31f-76ea9aac8046) |


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
The screenshots look good to me! @evykassirer are you up for reviewing this one? I haven't tested.

evykassirer commented:
Yay thanks for working on this! Once conflicts are resolved I'll check it locally and take a look.

reharsh commented:
@evykassirer I've resolved the conflicts, please check it out when you get time. thanks

evykassirer commented:
I'm wondering if we can address this TODO now:

```css
/* TODO: Once all layouts using this button
   are modernized, the Font Awesome icon
   should be replaced with a Zulip icon,
   and its formatting should have no extra
   space around its viewbox in SVG. */
.clear_search_button {
```

I also found one remaining instance of `<i class="fa fa-remove" aria-hidden="true"></i>` inside a `clear_search_button` div, in `set_status_overlay` -- can you add that to this PR?

evykassirer commented on web/templates/filter_topics.hbs:
```diff
@@ -1,6 +1,6 @@
     <input class="topic-list-filter home-page-input filter_text_input" id="filter-topic-input" type="text" autocomplete="off" placeholder="{{t 'Filter topics'}}" />
     <button type="button" class="bootstrap-btn clear_search_button" id="clear_search_topic_button">
-        <i class="fa fa-remove" aria-hidden="true"></i>
+        <i class="zulip-icon zulip-icon-close" aria-hidden="true"></i>
```
This one is looking a little high 

<img width="303" alt="image" src="https://github.com/user-attachments/assets/a66cef65-337e-4316-9d4b-2f8ad0bafef7">

evykassirer reviewed (commented):
This repeats the same style several times -- I'm wondering if we could add a classname that all of them use to only have to write most of these styles out once. Maybe we can use the existing `clear_search_button` in this way?

reharsh commented:
Hello @evykassirer,  

I've updated the `clear_search_button` class to apply common styles for clear buttons, reducing repetition. The '**x**' button in the `user_status_overlay` has been updated to use this class. Additionally, the right sidebar search box now shares the same class, and alignment issues have been fixed. Please take a look!

evykassirer commented on web/styles/app_components.css:
```diff
@@ -407,32 +407,27 @@ input.settings_text_input {
-   are modernized, the Font Awesome icon
-   should be replaced with a Zulip icon,
-   and its formatting should have no extra
-   space around its viewbox in SVG. */
```
Probably wait to remove this until the last commit once all the icons are replaced?

evykassirer replied:
You should now remove this comment in the last commit!

evykassirer commented on web/styles/app_components.css:
```diff
@@ -471,16 +466,6 @@ input.settings_text_input {
-    & input {
-        padding-right: 20px;
-    }
-}
```
Why is topic list code changed in the settings commit? If it's because you're editing `clear_search_button`, I would suggest doing something so you can keep the commits more separated. One option would be temporarily renaming `clear_search_button` to `old_clear_search_button` everywhere, then start a new `clear_search_button` with just the styles you're keeping and apply that to each close button as you migrate each of them, and then delete `old_clear_search_button` at the end.

evykassirer commented on web/styles/left_sidebar.css:
```diff
@@ -1738,11 +1734,9 @@ li.topic-list-item {
-        /* Use the border-box model so flex
-           can do its thing despite whatever
-           padding and border we specify. */
+        /* Use the border-box model */
```
This comment seems useless, you could just delete it.

evykassirer reviewed (commented):
I love the reuse of code! Getting the commits clean might involve a bit more work, left a comment about that.

I'm also noticing existing close buttons like in the search bar are darker:

![image](https://github.com/user-attachments/assets/230d7c3b-3f49-48b5-958f-708ced514117)

![image](https://github.com/user-attachments/assets/6b60dd6c-699d-421c-a8fa-7760d022cd6b)

Do you know why the new ones are lighter grey? Having them consistent would be better unless there's a good reason to not do that.

reharsh commented:
@evykassirer Thank you so much for the appreciation:)

I've made all the icons a bit darker to match the others, adjusted the comments, and cleaned up the commit history.  
Could you please take another look? Thanks!

evykassirer commented on web/styles/app_components.css:
```diff
@@ -412,7 +412,7 @@ input.settings_text_input {
    and its formatting should have no extra
    space around its viewbox in SVG. */
-.clear_search_button {
+.old_clear_search_button {
```
Add a bit more detail to the commit message for this rename, to make it clear this is temporary

evykassirer replied:
You wrote

> This temporary change preserves commit history while migrating all 'x'
> icons in search boxes to use a unified clear_search_button style.

I'd instead say:

> This temporary change allows us to preserve these old styles while iteratively
> move all the close buttons to new styles. Once the migration is complete,
> `old_clear_search_button` won't exist anymore and there will be new styles
> for `clear_search_button`.

evykassirer commented on web/styles/subscriptions.css:
```diff
@@ -542,10 +539,8 @@ h4.user_group_setting_subsection_title {
-    grid-area: clear-search;
-    padding: 0;
+#search_group_name:placeholder-shown + #clear_search_group_name {
+    visibility: hidden;
```
Can you put this in a separate commit to make it more clear that it's not part of the conversion to the new icon and more of a missing feature from before?

evykassirer commented on web/styles/user_status.css:
```diff
@@ -3,14 +3,18 @@
+        grid-template:
+            "status-icon search-input clear-search" auto / auto minmax(0, 1fr)
+            auto;
+        align-items: center; /* Vertically center items */
```
Unnecessary comment (though I agree it's often hard to remember which is which)

evykassirer commented on web/styles/left_sidebar.css:
```diff
@@ -1753,41 +1746,23 @@ li.topic-list-item {
+
+    .stream-list-filter:placeholder-shown + #clear_search_stream_button {
+        display: none;
+    }
```
Same here, put in separate commit.

evykassirer commented on web/styles/popovers.css:
```diff
@@ -631,16 +631,15 @@ ul.popover-group-menu-member-list {
-        #clear_stream_search {
-            display: none;
-            grid-area: clear-search;
+        .stream-search:placeholder-shown + #clear_stream_search {
```
And here -- you could probably combine these into one shared commit for hiding the close button when there's no text present in the input?

evykassirer commented on web/styles/user_status.css:
```diff
@@ -23,9 +27,14 @@
+        }
+
+        #user-status-input:placeholder-shown + .clear_search_button {
+            visibility: hidden;
```
(same)

evykassirer reviewed (commented):
Really close! Just a few more comments :)

reharsh commented:
@evykassirer, these are done. please let me know if further changes are needed.

evykassirer commented:
Looks good! Please update that one commit message, but otherwise this looks ready for @timabbott to do integration review. Thanks again for working on this!

timabbott commented on web/styles/right_sidebar.css:
```diff
@@ -482,6 +482,10 @@ $user_status_emoji_width: 24px;
+
+        .user-list-filter:placeholder-shown + #clear_search_people_button {
+            display: none;
+        }
```
Why is this one changing `display`, unlike `visibility` for the others?

timabbott commented:
This seems solid, merged, thanks @reharsh and @evykassirer! 

Can you post about https://github.com/zulip/zulip/pull/32638#discussion_r1906215682 in #frontend, rather than replying here? If we should standardize one `visibility` or `display`, we can do it then.

karlstolley commented on web/styles/left_sidebar.css:
```diff
@@ -1142,12 +1134,10 @@ li.top_left_scheduled_messages {
-        [clear-button-start] var(--left-sidebar-vdots-width)
-        [clear-button-end filter-box-end] 0;
+    padding: 8px 10px;
+    display: grid;
```
Okay. So we don't want to do this. The deleted `grid-template-columns` above are so we can keep all the sidebar rows sharing roughly the same grid areas. You should just be able to reference the `clear-button` area if you need to more explicitly place the clear icon, and there should be no need to have to set padding here, either.

evykassirer replied:
Thanks for the review and fix Karl!
